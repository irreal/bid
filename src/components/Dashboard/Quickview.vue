<template>
  <v-layout align-end justify-end py-0 class="headline">
    <v-card class="pa-3 elevation-0" color="transparent" dark height="150px">
      <v-card-title>
        {{ weather.location }}
      </v-card-title>
      <v-card-title>
        <span class="text-light">Lokacija</span>
      </v-card-title>
    </v-card>
    <v-card class="pa-3 elevation-0" color="transparent" dark height="150px">
      <v-card-title>
        <span>{{ weather.temperature }}</span>
      </v-card-title>
      <v-card-title>
        <span class="text-light">Temperatura</span>
      </v-card-title>
    </v-card>
    <v-card class="pa-3 elevation-0" color="transparent" dark height="150px">
      <v-card-title class="py-0 pt-1 mb-2">
        <img :src="weather.icon" />
      </v-card-title>
      <v-card-title>
        <span class="text-light">Vreme</span>
      </v-card-title>
    </v-card>
  </v-layout>
</template>
<script>
import axios from "axios";
export default {
  name: "QuickView",
  data() {
    return {
      weather: {
        temperature: "--°",
        location: "nepoznato",
        icon: ""
      }
    };
  },
  mounted() {
    axios
      .get("/.netlify/functions/weather")
      .then(resp => {
        this.weather.temperature = Math.round(resp.data.main.temp) + "°";
        this.weather.location = resp.data.name;
        this.weather.icon = `https://openweathermap.org/img/w/${
          resp.data.weather[0].icon
        }.png`;
      })
      .catch(err => {
        console.error("Error getting weather details", err);
      });
  }
};
</script>
<style scoped lang="scss">
.fixed-height {
  height: 64px;
}
</style>
