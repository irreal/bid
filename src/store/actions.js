export default {
  updateUser: ({ commit }, payload) => {
    commit("setUser", payload.currentUser);
  }
};
