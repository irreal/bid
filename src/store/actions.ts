import { ActionTree } from 'vuex';
import State from './state';

const actions: ActionTree<State, State> = {
    //   updateColor({ commit }: ActionContext<State, State>, newColor: string): void {
    //     commit(COLOR, newColor);
    //   },
    };

export default actions;
