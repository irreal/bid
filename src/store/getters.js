export default {
  getUserStatus: state => !!state.user,
  getUser: state => state.user
};
