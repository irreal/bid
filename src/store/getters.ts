import { GetterTree } from 'vuex';
import State from './state';
import Plan from '@/models/plan';

const getters: GetterTree<State, State> = {
    plans(state: State): Plan[] {
      return state.plans;
    },
    // colour(state: State): string {
    // 	return state.color;
    // },
  };

export default getters;
