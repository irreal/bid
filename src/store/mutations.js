export default {
  setUser: (state, currentUser) => {
    state.user = currentUser;
  }
};
