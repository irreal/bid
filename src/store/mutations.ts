import { MutationTree } from 'vuex';
import State from './state';

const mutations: MutationTree<State> = {
    removePlans(state: State) {
      state.plans = [];
    },
    // color(state: State, newColor: string): void {
    // 	state.color = newColor;
    // }
  };

export default mutations;
