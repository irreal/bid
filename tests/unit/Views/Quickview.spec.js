import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Quickview from "@/components/Dashboard/Quickview.vue";
import Vuetify from "vuetify";
import Vue from "vue";
import initialState from "@/store/state";
import actions from "@/store/actions";
import getters from "@/store/getters";
import mutations from "@/store/mutations";
import userFixture from "../fixtures/user";
import weatherFixture from "../fixtures/weather";
import axios from "axios";
import flushPromises from "flush-promises";

const localVue = createLocalVue();
localVue.use(Vuex);
Vue.use(Vuetify);

describe("Quickview View", () => {
  let state;

  const defaultStoreObject = () => ({
    state,
    actions,
    getters,
    mutations
  });
  const build = (storeObject = defaultStoreObject()) => {
    const store = new Vuex.Store(storeObject);

    const quickview = shallowMount(Quickview, {
      localVue,
      store
    });

    return {
      quickview,
      store
    };
  };

  beforeEach(() => {
    state = { ...initialState };
  });

  it("renders the component", () => {
    // arrange
    const { quickview } = build();
    // assert
    expect(quickview.html()).toMatchSnapshot();
  });

  it("renders the component with a logged in user", () => {
    // arrange
    state.user = userFixture;
    const { quickview } = build();
    // assert
    expect(quickview.html()).toMatchSnapshot();
  });

  it("calls the weather lambda and displays results", async () => {
    jest.mock("axios");
    axios.get = jest.fn(() => {
      return Promise.resolve({ data: weatherFixture });
    });
    const { quickview } = build();
    await flushPromises();
    expect(axios.get).toBeCalledWith("/.netlify/functions/weather");

    expect(quickview.vm.weather.temperature).toEqual("8°");
    expect(quickview.vm.weather).toMatchSnapshot();
  });

  it("calls the weather lambda and displays nothing if call fails", async () => {
    jest.mock("axios");
    axios.get = jest.fn(() => {
      return Promise.reject(new Error("fake error"));
    });
    global.console = { error: jest.fn() };
    const { quickview } = build();
    await flushPromises();

    expect(axios.get).toBeCalledWith("/.netlify/functions/weather");

    expect(global.console.error).toBeCalledWith(
      "Error getting weather details",
      new Error("fake error")
    );

    expect(quickview.vm.weather.temperature).toEqual("--°");
    expect(quickview.vm.weather).toMatchSnapshot();
  });
});
