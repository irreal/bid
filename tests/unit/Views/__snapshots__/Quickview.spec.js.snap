// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quickview View calls the weather lambda and displays nothing if call fails 1`] = `
Object {
  "icon": "",
  "location": "nepoznato",
  "temperature": "--째",
}
`;

exports[`Quickview View calls the weather lambda and displays results 1`] = `
Object {
  "icon": "https://openweathermap.org/img/w/10d.png",
  "location": "Belgrade",
  "temperature": "8째",
}
`;

exports[`Quickview View renders the component 1`] = `
<v-layout-stub tag="div" align-end="" justify-end="" py-0="" class="headline">
  <v-card-stub tag="div" color="transparent" height="150px" dark="true" class="pa-3 elevation-0">
    <v-card-title-stub>
      nepoznato
    </v-card-title-stub>
    <v-card-title-stub><span class="text-light">Lokacija</span></v-card-title-stub>
  </v-card-stub>
  <v-card-stub tag="div" color="transparent" height="150px" dark="true" class="pa-3 elevation-0">
    <v-card-title-stub><span>--째</span></v-card-title-stub>
    <v-card-title-stub><span class="text-light">Temperatura</span></v-card-title-stub>
  </v-card-stub>
  <v-card-stub tag="div" color="transparent" height="150px" dark="true" class="pa-3 elevation-0">
    <v-card-title-stub class="py-0 pt-1 mb-2"><img src=""></v-card-title-stub>
    <v-card-title-stub><span class="text-light">Vreme</span></v-card-title-stub>
  </v-card-stub>
</v-layout-stub>
`;

exports[`Quickview View renders the component with a logged in user 1`] = `
<v-layout-stub tag="div" align-end="" justify-end="" py-0="" class="headline">
  <v-card-stub tag="div" color="transparent" height="150px" dark="true" class="pa-3 elevation-0">
    <v-card-title-stub>
      nepoznato
    </v-card-title-stub>
    <v-card-title-stub><span class="text-light">Lokacija</span></v-card-title-stub>
  </v-card-stub>
  <v-card-stub tag="div" color="transparent" height="150px" dark="true" class="pa-3 elevation-0">
    <v-card-title-stub><span>--째</span></v-card-title-stub>
    <v-card-title-stub><span class="text-light">Temperatura</span></v-card-title-stub>
  </v-card-stub>
  <v-card-stub tag="div" color="transparent" height="150px" dark="true" class="pa-3 elevation-0">
    <v-card-title-stub class="py-0 pt-1 mb-2"><img src=""></v-card-title-stub>
    <v-card-title-stub><span class="text-light">Vreme</span></v-card-title-stub>
  </v-card-stub>
</v-layout-stub>
`;
